docs/notebooks/inequality_constraints.pct.py

```diff
@@ -146,6 +148,105 @@ def create_bo_model(data):
 )
 plt.show()
 
+# %% [markdown]
+# ## Define a batch acquisition rule
+#
+# It is sometimes beneficial to query several points at a time instead of one. We show here how to create ad-hoc extensions of the previous acquistion function and rule.
+
+# %%
+from trieste.acquisition.function import BatchAcquisitionFunctionBuilder, BatchReparametrizationSampler
+
+
+class BatchExpectedConstrainedImprovement(BatchAcquisitionFunctionBuilder):
+    def __init__(
+        self,
+        objective_tag,
+        constraint_builder,
+        sample_size,
+        threshold,
+    ):
+        self._objective_tag = objective_tag
+        self._constraint_builder = constraint_builder
+        self._min_feasibility_probability = 0.5
+        self._sample_size = sample_size
+        self._jitter = 1e-6
+        self.threshold = threshold
+
+    def prepare_acquisition_function(self, datasets, models):
+        objective_model = models[self._objective_tag]
+        objective_dataset = datasets[self._objective_tag]
+
+        constraint_fn = self._constraint_builder.prepare_acquisition_function(datasets, models)
+        pf = constraint_fn(objective_dataset.query_points)
+        is_feasible = pf >= self._min_feasibility_probability
+
+        if not tf.reduce_any(is_feasible):
+            return constraint_fn
+
+        mean, _ = objective_model.predict(objective_dataset.query_points)
+        eta = tf.reduce_min(tf.boolean_mask(mean, is_feasible), axis=0)
+
+        samplers = {tag: BatchReparametrizationSampler(self._sample_size, models[tag]) for tag in models}
+
+        def batch_efi(at):
+            samples = {tag: samplers[tag].sample(at, jitter=self._jitter) for tag in models}
+            objective_samples = tf.squeeze(samples["OBJECTIVE"], axis=-1)  # [N, S, B]
+            constraint_samples = tf.squeeze(samples["CONSTRAINT"], axis=-1)  # [N, S, B]
+            feasible_mask = constraint_samples < self.threshold
+            improvement = tf.where(feasible_mask, tf.math.maximum(eta - objective_samples, 0.), 0.)
+            batch_improvement = tf.math.reduce_max(improvement, axis=-1)  # [N, S]
+            return tf.reduce_mean(batch_improvement, axis=-1, keepdims=True)  # [N, 1]
+
+        return batch_efi
+
+
+num_query_points = 4
+pof = trieste.acquisition.ProbabilityOfFeasibility(threshold=Sim.threshold)
+batch_eci = BatchExpectedConstrainedImprovement(OBJECTIVE, pof.using(CONSTRAINT), 50, Sim.threshold)
```

Comment:
will pof work in a batch builder?

Reply:
actually I removed pof so this won't apply